platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/bls/protocol/RandomGroupElements.java

```diff
@@ -0,0 +1,141 @@
+    private final GroupElement randomGroupElement1;
+
+    /** The second random group element, in the public key group of the bilinear map */
+    private final GroupElement randomGroupElement2;
+
+    /**
+     * The underlying bilinear map. The random elements are part of the key group of this bilinear
+     * map
+     */
+    private final BilinearMap bilinearMap;
+
+    /**
+     * Constructor that generates new random group elements
+     *
+     * @param bilinearMap the underlying bilinear map
+     * @param random a source of randomness
+     */
+    public RandomGroupElements(final BilinearMap bilinearMap, final Random random) {
+        this.randomGroupElement1 = bilinearMap
+                .keyGroup()
+                .randomElement(randomByteArray(random, 32))
+                .compress();
+
+        this.randomGroupElement2 = bilinearMap
+                .keyGroup()
+                .randomElement(randomByteArray(random, 32))
+                .compress();
+
+        this.bilinearMap = bilinearMap;
+    }
+
+    /**
+     * Constructor for when random group elements have already been generated. Both random elements
+     * must be in the public key group of the bilinear map.
+     *
+     * @param bilinearMap the underlying bilinear map
+     * @param randomGroupElement1 the first random group element
+     * @param randomGroupElement2 the second random group element
+     */
+    public RandomGroupElements(
+            final BilinearMap bilinearMap,
+            final GroupElement randomGroupElement1,
+            final GroupElement randomGroupElement2) {
+
+        assertPublicKeyGroupMembership(bilinearMap, randomGroupElement1, "randomGroupElement1");
+        assertPublicKeyGroupMembership(bilinearMap, randomGroupElement2, "randomGroupElement2");
+
+        this.randomGroupElement1 = randomGroupElement1;
+        this.randomGroupElement2 = randomGroupElement2;
+
+        this.bilinearMap = bilinearMap;
+    }
+
+    /**
+     * Gets the first random group element
+     *
+     * @return {@link #randomGroupElement1}
+     */
+    public GroupElement getRandomGroupElement1() {
+        return randomGroupElement1;
+    }
+
+    /**
+     * Gets the second random group element
+     *
+     * @return {@link #randomGroupElement2}
+     */
+    public GroupElement getRandomGroupElement2() {
+        return randomGroupElement2;
+    }
+
+    /**
+     * Gets the underlying bilinear map
+     *
+     * @return the bilinear map
+     */
+    public BilinearMap getBilinearMap() {
+        return bilinearMap;
+    }
+
+    /**
+     * Generates a commitment to the random group elements
+     *
+     * @return the generated commitment
+     */
+    public byte[] commit() throws NoSuchAlgorithmException {
+        MessageDigest digest = MessageDigest.getInstance("SHA-256");
+
+        digest.update(randomGroupElement1.toBytes());
+        digest.update(randomGroupElement2.toBytes());
+
+        return digest.digest();
```

Comment:
Would make more elegant API if you caught the `NoSuchAlgorithmException` here and re-threw it as something you don't need to catch

Reply:
I would pass in the digest in the constructor and reuse the same instance, that avoid the exception altogether

Reply:
Do we need to be able to support other levels besides SHA-256?  I'm wondering about whether or not it should be hard coded, or loaded from configuration from start of day.  If it was loaded at start of day, it could be kept as a static retrievable from a global settings.

Reply:
In general I would move the "SHA-256" string to a constant and have a MessageDigestFactory. By doing so we can make it easily configurable later.